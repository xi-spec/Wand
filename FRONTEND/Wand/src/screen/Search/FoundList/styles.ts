import { StyleSheet } from 'react-native';
import { SIZE, FONTS, COLORS } from '../../../constants/theme';

const styles = StyleSheet.create({
  listContainer: {
    margin: 10
  },
  detail: {
    fontSize: SIZE.second,
    fontFamily: FONTS.light,
    marginBottom: 5
  },
  imageContainer: {
    margin: 5
  },
  image: {
    justifyContent: 'center',
    width: 190,
    height: 300,
    borderRadius: 5,
    marginBottom: 10
  },
  cost: {
    fontFamily: FONTS.primary,
    fontSize: SIZE.second,
    color: COLORS.second,
    marginBottom: 10
  }
});

export default styles;
