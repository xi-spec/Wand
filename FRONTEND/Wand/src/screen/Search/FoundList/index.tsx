import React from 'react';
import { Product } from '../../../models/initiaState';
import { Text, TouchableOpacity, FlatList, Image } from 'react-native';
import styles from './styles';
import setCostValue2Decimals from '../../../utils/setCostValue2Decimals';

export default function FoundList ({ navigation, foundList }:{navigation:any, foundList:Product[]}) {
  return (
<FlatList
        bounces={false}
        style={styles.listContainer}
        data={foundList}
        numColumns={2}
        keyExtractor={(item) => item._id}
        renderItem={({ item }) =>
        <TouchableOpacity
        style={styles.imageContainer}
        key={item.name}
        testID='foundListImage'
        onPress={() => navigation.navigate('Detail', { productId: item._id })}>
      <Image source={{ uri: item.images.cover }}
      style={styles.image}>
        </Image>
        <Text style={styles.detail}>{item.name}</Text>
        <Text style={styles.cost}>{`${setCostValue2Decimals(item.cost.value)} ${item.cost.currency}`}</Text>
      </TouchableOpacity>
        }>
      </FlatList>
  );
}
