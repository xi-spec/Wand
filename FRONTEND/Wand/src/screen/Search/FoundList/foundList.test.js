
import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import FoundList from './index';

describe('Given a foundList component', () => {
  let navigation;
  let foundList;

  beforeEach(() => {
    navigation = {
      navigate: jest.fn()
    };
    foundList = [{
      _id: '2',
      name: 'be happy 1',
      cost: { value: 2, currency: 'euro' },
      images: {
        alls: ['url'],
        cover: 'url'
      }
    }];
  });
  describe('When is rendered', () => {
    test('Then is matched to snapshop', () => {
      const tree = render(<FoundList foundList={foundList} />);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When text of search input is change with "HELLO" ', () => {
    test('Then text of search input  to be "HELLO"', () => {
      const { getByTestId } = render(<FoundList foundList={foundList} navigation={navigation}/>);

      fireEvent.press(getByTestId('foundListImage'));

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
