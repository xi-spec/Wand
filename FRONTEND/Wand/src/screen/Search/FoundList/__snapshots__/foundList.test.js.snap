// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a foundList component When is rendered Then is matched to snapshop 1`] = `
<RCTScrollView
  bounces={false}
  data={
    Array [
      Object {
        "_id": "2",
        "cost": Object {
          "currency": "euro",
          "value": 2,
        },
        "images": Object {
          "alls": Array [
            "url",
          ],
          "cover": "url",
        },
        "name": "be happy 1",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "margin": 10,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "margin": 5,
              "opacity": 1,
            }
          }
          testID="foundListImage"
        >
          <Image
            source={
              Object {
                "uri": "url",
              }
            }
            style={
              Object {
                "borderRadius": 5,
                "height": 300,
                "justifyContent": "center",
                "marginBottom": 10,
                "width": 190,
              }
            }
          />
          <Text
            style={
              Object {
                "fontFamily": "RalewayLight",
                "fontSize": 14,
                "marginBottom": 5,
              }
            }
          >
            be happy 1
          </Text>
          <Text
            style={
              Object {
                "color": "#F3603F",
                "fontFamily": "MontserratRegular",
                "fontSize": 14,
                "marginBottom": 10,
              }
            }
          >
            2.00 euro
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
